superset/utils/pandas_postprocessing.py

```diff
@@ -0,0 +1,323 @@
+        options = agg_obj.get("options", {})
+        agg_funcs[name] = NamedAgg(column=column, aggfunc=partial(func, **options))
+
+    return agg_funcs
+
+
+def _append_columns(
+    base_df: DataFrame, append_df: DataFrame, columns: Dict[str, str]
+) -> DataFrame:
+    """
+    Function for adding columns from one DataFrame to another DataFrame. Calls the
+    assign method, which overwrites the original column in `base_df` if the column
+    already exists, and appends the column if the name is not defined.
+
+    :param base_df: DataFrame which to use as the base
+    :param append_df: DataFrame from which to select data.
+    :param columns: columns on which to append, mapping source column to
+           target column. For instance, `{'y': 'y'}` will replace the values in
+           column `y` in `base_df` with the values in `y` in `append_df`,
+           while `{'y': 'y2'}` will add a column `y2` to `base_df` based
+           on values in column `y` in `append_df`, leaving the original column `y`
+           in `base_df` unchanged.
+    :return: new DataFrame with combined data from `base_df` and `append_df`
+    """
+    return base_df.assign(
+        **{
+            target: append_df[append_df.columns[idx]]
+            for idx, target in enumerate(columns.values())
+        }
+    )
+
+
+def pivot(  # pylint: disable=too-many-arguments
+    df: DataFrame,
+    index: List[str],
+    columns: List[str],
+    aggregates: Dict[str, Dict[str, Any]],
+    metric_fill_value: Optional[Any] = None,
+    column_fill_value: Optional[str] = None,
+    drop_missing_columns: Optional[bool] = True,
+    combine_value_with_metric=False,
+    marginal_distributions: Optional[bool] = None,
+    marginal_distribution_name: Optional[str] = None,
+) -> DataFrame:
+    """
+    Perform a pivot operation on a DataFrame.
+
+    :param df: Object on which pivot operation will be performed
+    :param index: Columns to group by on the table index (=rows)
+    :param columns: Columns to group by on the table columns
+    :param metric_fill_value: Value to replace missing values with
+    :param column_fill_value: Value to replace missing pivot columns with
+    :param drop_missing_columns: Do not include columns whose entries are all missing
+    :param combine_value_with_metric: Display metrics side by side within each column,
+           as opposed to each column being displayed side by side for each metric.
+    :param aggregates: A mapping from aggregate column name to the the aggregate
+           config.
+    :param marginal_distributions: Add totals for row/column. Default to False
+    :param marginal_distribution_name: Name of row/column with marginal distribution.
+           Default to 'All'.
+    :return: A pivot table
+    :raises ChartDataValidationError: If the request in incorrect
+    """
+    _validate_columns(df, index)
+    _validate_columns(df, columns)
+    _validate_columns(df, aggregates.keys())
```

Comment:
Can probably refactor this into a decorator:

```python

def validate_column_args(*argnames: str) -> Callable:
    def wrapper(fn):
        def wrapped(df, **options):
            columns = df.columns.tolist()
            for name in argnames:
                # iterate through a dict type is fine because it'll just iter through the keys
                if not all(elem in columns for elem in options[name]):
                    raise ChartDataValidationError(
                       _(
                          "Referenced columns not available in DataFrame. "
                           "Columns in DataFrame: %(df_cols)s. Referenced columns: %(columns)s",
                           df_cols=list(df.columns),
                           columns=list(columns),
                        )
                    )
            return fn(df, **options)
        return wrapped
    return wrapper

@validate_column_args('index', 'columns', 'aggregates')
def pivot(df: DataFrame, index: List[str], columns: List[str], ...):
    pass
```

Reply:
Good idea 👍